ipfx/data_set_features.py

```diff
@@ -68,6 +69,34 @@ def detection_parameters(stimulus_name):
     return DETECTION_PARAMETERS.get(stimulus_name, {})
 
 
+def record_errors(fn):
+    @functools.wraps(fn)
+    def wrapper(*args, **kwargs):
+        result, errors = fn(*args, **kwargs)
+        failed = (errors is not None)
+        return (
+            result,
+            {'failed_fx': failed,
+             'fail_fx_message': str(errors) if failed else None}
+            )
+    return wrapper
+
+
+def safe_fn(fvalue_on_failure=None):
```

Comment:
I suggest updating safe_fn to take an additional parameter, which specifies the subset of errors that are caught vs reraised.

Reply:
Also we might consider changing safe_fn to something more descriptive, like `fallback_value` (yes, I know `safe_fn` is my fault, sorry).